### PR TITLE
feat: add ECharts Pie chart

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Replace the existing NVD3 Pie chart with an ECharts based chart (see https://github.com/apache-superset/superset-ui/pull/772) and bump `superset-ui` packages to `0.15.3`. Also fixes color problems on boxplot and wordcloud.

### BEFORE
image

### AFTER
image
pie_anim

### ADDITIONAL INFORMATION

- [ ] Has associated issue: #10418 
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by JunlinC** [comment]
they look awesome!!thank you Ville!

File: tests/viz_tests.py [hunk 1]
```diff
@@ -1384,60 +1384,3 @@ def test_format_datetime_from_invalid_string(self):
...
-        data = pie_viz.get_data(self.base_df)
-        assert self.get_cols(data) == {"<NULL>", "1", "0.2", "0.3", "0.4"}
-
-    def test_multi_groupby(self):
-        datasource = self.get_datasource_mock()
-        pie_viz = viz.DistributionPieViz(
-            datasource, {"metrics": ["floatcol"], "groupby": ["intcol", "strcol"]},
-        )
-        data = pie_viz.get_data(self.base_df)
-        assert self.get_cols(data) == {"1, a", "2, b", "3, c", "4, <NULL>", "<NULL>, d"}
```
Comment (villebro) [inline tests/viz_tests.py:1443]: These have been moved to the plugin: https://github.com/apache-superset/superset-ui/blob/d0b6fa03488d732b24fe159a88f38194a3d350d2/plugins/plugin-chart-echarts/test/utils/series.test.ts#L87-L120

**Review by rusackas — APPROVED** [review]
Fantastic! Happy to see more of viz.py go away, and this first drop-in replacement for NVD3! And tests in the more sensible place! So many good things happening here.

@ktmud / @kristw Please double check that this PR (and ones like it) won't break any use cases you're aware of. I would assume people will be excited to see this one landing though.

**Review by pkdotson — APPROVED** [review]

**Review by mistercrunch — APPROVED** [review]

**Comment by ktmud** [comment]
Thank you! The new pie chart looks so much beautifuler!

**Comment by benceorlai** [comment]
Roadmap: https://github.com/apache-superset/superset-roadmap/issues/48

**Pull request merged**